src/main/java/hudson/scm/SubversionChangeLogBuilder.java

```diff
@@ -122,24 +122,20 @@ public boolean run(Collection<SubversionSCM.External> externals, Result changeLo
                 PathContext context = getUrlForPath(workspace.child(l.getLocalDir()), authProvider);
                 context.moduleWorkspacePath = l.getLocalDir();
                 changelogFileCreated |= buildModule(context, svnlc, logHandler);
+
+                // externals for this module location
+                List<SubversionSCM.External> externals = externalsMap.get(l.remote);
+                if (externals != null) {
+                  for (SubversionSCM.External ext : externals) {
+                    PathContext extContext = getUrlForPath(workspace.child(ext.path), authProvider);
+                    extContext.moduleWorkspacePath = ext.path;
+                    changelogFileCreated |= buildModule(extContext, svnlc, logHandler);
+                  }
+                }
             } finally {
                 manager.dispose();
             }
         }
-        ISVNAuthenticationProvider authProvider =
-                CredentialsSVNAuthenticationProviderImpl
-                        .createAuthenticationProvider(build.getParent(), scm, null, listener);
-        final SVNClientManager manager = SubversionSCM.createClientManager(authProvider).getCore();
-        try {
-            SVNLogClient svnlc = manager.getLogClient();
-            for(SubversionSCM.External ext : externals) {
```

Comment:
Is there any case in which this is still necessary as a fallback?

Reply:
before of this change, there was only one list of remotes now are classified by remote, so now only process external when there are externals, the removed login it is not needed anymore.

Reply:
Ok, seems fine to me then.